### DISCUSSION TITLE
How to enable chat support on Copilot in PyCharm IDE?

### DISCUSSION THREAD

**Discussion by augustocadini** [community/community#56901] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I just installed the Github Copilot in my PyCharm IDE but I can't find any tool to chat with it. It just shows me code suggestions. How to enable the Chat support, if is it available?
![Screen Shot 2023-05-31 at 23 02 08](https://github.com/community/community/assets/18659348/91de5c9f-3682-4e1f-b1be-f2a377bca5ae)

**Comment by Akash1134** [comment]
Hi @augustocadini ,

Thank you for posting in the Copilot community! We value your feedback and our team is paying attention to the feedback, feature requests & bugs we are hearing from users in the Community Discussions.

At this time, the public preview for Copilot for Chat is for VS or VSCode users. However, we'll share your feedback with the concerned team so that they can look into it.

**Reply by augustocadini** [reply]
Hi Akash, thank you for letting me know.

**Reply by eloyparedesq** [reply]
Please tell us when the plugin for PyCharm will be available, since it is a tool that a large number of developers use.

**Reply by kj-weber** [reply]
Hi there @Akash1134 - could you bring up this support request with the team again? It has been over 4 months and it appears it's still quite popular.

**Reply by m3i3k3e** [reply]
Hey @Akash1134,

I'd like to offer up a pretty good workaround that I discovered (and that may not be obvious to folks with a single IDE installed):

**One can simply run PyCharm with only the copilot extension while also running VSCode (accessing the chat function from the sidebar without ever leaving the welcome screen of VSCode).** You don't need to have a project open to access the chat extension in VSCode.

Now this IS assuming someone is willing to suffer through the dreaded ALT+TAB process detailed here: " (in PyCharm) CTRL+A, CTRL+C, ALT+TAB, (in VSCode) CTRL+V, ENTER, (have your mind blown), CTRL+C, ALT+TAB, CTRL+V (feel way smarter than you are), make this face: **:0**. "

It took me me so much longer to type that than it did to figure this out (jumping ahead of this excellent [Udemy Python fundamentals ](https://www.udemy.com/course/the-python-mega-course/)course):


```
waifus = []
rem_printed = False

while True:
    user_waifu = input("\033[1;33;40mWaifu List 1.0 - Type Add, Show, or Exit: \n\033[1;32;40m").lower()

    match user_waifu:
        case 'add':
            waifu = input("\033[1;33;40mName your Waifu: \n\033[1;32;40m")
            waifus.append(waifu)
        case "show":
            print(waifus)

            if "rem".lower() in [waifu.lower() for waifu in waifus] and not rem_printed:
                print("""\033[1;33;40mYou have EXCELLENT taste in waifus! \n\033[1;32;40m
                .                   ::     .     ..  ;@@  .@@@@ @@@@@@@         
                ,                    rr    .r;;,  ,,  @@@ .:@@@@ @@@@@@@@       
   .,          .:                 .   ir     i2; . .:.  ;,,.;@@@@ r@@@@@@@      
   :,          .:       .         :   :ss     :.  ...;,. ,,  r@@@A,2@@@@@@@     
   ;           ;.       .              s;;    .,.     ; ;, :  r@@@:. .G@@@@@.   
   r.          r                       ;r;;:    ,.:r:  ,. :,:. i:::  :; @@@@@ 
   r.          S                 .     ,s2;;:      :S33;   ,,, .5.:;,....@@@@@
  .:,         is           .            ssir;;        .rXS: ... ,i....  , ;@@@  
  ,::         @            :     .      ssrisr;     ,.    :r,... ,. r::,  ,@;@i 
  ;::        ,@           ..     .      iirrsis;   ..         ...  ,:   .. ,X2r.
  ;;:.       @,           r      .   .. 2XS25XB5s.    ,,        .. .:        r32
  rr,,       @           .s     ,.  ... ;,.   .:r:  .: :r           ,          s
 .;s,:      A;    .      ,.     :  ..       .,:;::,  .. :r   ,    . ..          
 .ri,:      @     .     .      .:  ,,.: ii@@@@@@@@@s:.   :,  .;   .  ,        ;.
 .ri::     &.           .      ,,  :.:, 9@@@@@@@@@@@2:,   ;.  ;r   .  .  .    : 
 ,ri;,.   ;9           ,  .    ;   :.:. @@@@@@@@@@@@@@r   ,,   i:  ..  .  .   .;
 .rsr.,   @  .        &.  .   :, : ,.:  @@@@@@@@@@@@@@@s  B,   ,,  :,.     .  :9
 ,rri,:  X;  .       ::      .: r : ,; ,@@@@@@@@@@@@i@@@; @@s:..   :;,    . .   
 ,sS#:  .2  ,,      .       ,: ;s , :; X@@@@@@@@@@@@ @@@@ 5@@@rr   :r.     .   .
 :@s   :X   ,      .       .: :@.: .r  @@@@@@@@@@is   .,s::@@@@3,  ;r.     .  :,
 :. ,@,@,   ,   . .  .    ,r ,#r.. ;; ;@@@@@@;            .r@@@@:  ;r         .:
  ;@@@@r    ,   r:  .    .s ,3X , :s  @@@@@              ,  , ;@;  rs      :,   
 .@@@@@,.   ,  rr  .:.   r ,i# . ,s: @@@@: ,     :,      ;G   @@r  ir  .   .i  .
  @@@@@@,   .  :,. :;.  r..r@   .ii ,@@@@:s  SX.@@3.      5@3  @X  X, .,    r.  
  @@@@@@G.  ;  ;;:;r;  i, r@.   si  @@@@@@  r@r..   .      @@@@ ;  X ..;    ,;  
  @@G@@@@5 .:  r;rrr; :; r@,   ri  @@@@@@@r@@@@     @r     @@@@:  .;.,,.    r;  
,,2@@  i@@; :  ,:,:r, r i@,   rr  @@@@@@@@@@@@@    ,;@ @s,.@@@@:  ;.S ,   @@@@@ 
i  @@@:     ,     ,, ; s@    :;..@@@@@@@@@@@@@@@  B@@;@@@  @@@@@r ,sS . r@@; @@@
 . @@S.      . : .rs,.;#    :;::@@@@@@@@@@@@@@@@,   ,Sr,  @@@r@@  r@. ,@@@:   @@
 ; :             :2:  X  .A5;:;@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@; i@S  @@@@ @@ ;@
 .:   . 2rr      ,:  .,.@@;.;@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@5 :@X  .@@@  :@@@@
  ,  H,M5         :&i;@@@;s@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@;  @s.  @@@  @@@@@@
 ,. .3i      X     @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@#@@@ i@; ,@; : @@@@ @@@@@@@
 .  .;@      3:    @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@: @@@@.  ,r  @:@@@@S@@@@@@@@
    ..@@   @@rs,   @@@@@@@@@@@@@@@ @@@@@@@@@@@@@@@@@@ @@@  .;:,H@2@@@@@@@@@@@@@@
,.  : .@@  r@@;;;  @@@@@@@@@@@@@@@ #@@@@@@@@@@@@@@@@@@@@@X@@@@@@@@@@@@@@@@@@@@@@
::. 3@ ,@@:   @@@:@@@@@@@@@@@@@@@@@#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ 
 :;. 9;  G@;#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@.@@@@@@  
  .;. Ss  ,@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@  3
    ,.,@@:  .,X@@@@ @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@  :i
        i@@2;h 3@@@ H@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@5  .s  
;,        r,5@ ,@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ ;B: @@, .
,:r;;:.   , #A,,##@@@@@@@@@@@@@@@@@@@@B&@@#:       :@@@@@@@@@@@@@@@@@@ @2 @@,   
.  ,2@@@@@;:Ai@##@@@@@@@@@@@@@@@@@i2@@@@H;.  .,..,  @@@@@@@@@@@@@@@@@@,:r@@    .
sS:  G@@@@9###M@@@@@@@@@@@@@@@@@   ;:.     ... ...  @@@@@@@@@@@@@@@@@:;@@       
s :;.:@@@@@@@@@@@@@@@@@@@@@@@@   ..    ,;;riirr;;r @@@@@@@@@@@@@@@@@;.A5  A..  ,
@ .:. &@@@@@@@@@@@@@@@@@@@@@:  ,,:..:r@@@@@@@@@@@.i@@@@@@@@@@@@@@@@  i5; @@:   X
@;,::   .@@@@@@@@@@@@@@@@@@@   ,,.;h@@BAHHHHB@H, @@@@@@@@@@@@@@@@@h i5r ,@@    @
;@@h:r.. ,@@@@@@@@@@@@@@@@@@@i   S@@@AAAGG32r,,@@@@@@@@@@@@@@@@@@: rrr  @@@ . 5@
  r@@@@@@@ @@@@@@@@@@@@@@@@@@@@@:.,,:;339G#@@@@@@@@@@@@@@@@@@@@@  ,:;  @@@    @ 
s    ;ri@@; :@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@. ,:::, .@@.   .  
i@            @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@.  i@i:.  @@@       
 ;@;   .       r@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@:   @@ ;,  @@@        
   @@   r,       r@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@    .@@   @@@@@,       :
 s, ;@r  sX.        ;@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@2 ;   @@;   @@@@@@@@@@   ;S
 5s;..;i  ,@9,           ,@@@@@@@@@@@@@@@@@@@@@@@@@,  @@ r@@    @@@@@@@@@@@@@h.,
                """)

                rem_printed = True

            elif "rem".lower() not in [waifu.lower() for waifu in waifus]:
                rem_printed = False

            else:
                pass

        case 'exit':
            exit()
```

**Reply by itsDrew86** [reply]
> Hey @Akash1134,
> 
> 
> 
> I'd like to offer up a pretty good workaround that I discovered (and that may not be obvious to folks with a single IDE installed):
> 
> 
> 
> **One can simply run PyCharm with only the copilot extension while also running VSCode (accessing the chat function from the sidebar without ever leaving the welcome screen of VSCode).** You don't need to have a project open to access the chat extension in VSCode.
> 
> 
> 
> Now this IS assuming someone is willing to suffer through the dreaded ALT+TAB process detailed here: " (in PyCharm) CTRL+A, CTRL+C, ALT+TAB, (in VSCode) CTRL+V, ENTER, (have your mind blown), CTRL+C, ALT+TAB, CTRL+V (feel way smarter than you are), make this face: **:0**. "
> 
> 
> 
> It took me me so much longer to type that than it did to figure this out (jumping ahead of this excellent [Udemy Python fundamentals ](https://www.udemy.com/course/the-python-mega-course/)course):
> 
> 
> 
> 
> 
> ```
> 
> waifus = []
> 
> rem_printed = False
> 
> 
> 
> while True:
> 
>     user_waifu = input("\033[1;33;40mWaifu List 1.0 - Type Add, Show, or Exit: \n\033[1;32;40m").lower()
> 
> 
> 
>     match user_waifu:
> 
>         case 'add':
> 
>             waifu = input("\033[1;33;40mName your Waifu: \n\033[1;32;40m")
> 
>             waifus.append(waifu)
> 
>         case "show":
> 
>             print(waifus)
> 
> 
> 
>             if "rem".lower() in [waifu.lower() for waifu in waifus] and not rem_printed:
> 
>                 print("""\033[1;33;40mYou have EXCELLENT taste in waifus! \n\033[1;32;40m
> 
>                 .                   ::     .     ..  ;@@  .@@@@ @@@@@@@         
> 
>                 ,                    rr    .r;;,  ,,  @@@ .:@@@@ @@@@@@@@       
> 
>    .,          .:                 .   ir     i2; . .:.  ;,,.;@@@@ r@@@@@@@      
> 
>    :,          .:       .         :   :ss     :.  ...;,. ,,  r@@@A,2@@@@@@@     
> 
>    ;           ;.       .              s;;    .,.     ; ;, :  r@@@:. .G@@@@@.   
> 
>    r.          r                       ;r;;:    ,.:r:  ,. :,:. i:::  :; @@@@@ 
> 
>    r.          S                 .     ,s2;;:      :S33;   ,,, .5.:;,....@@@@@
> 
>   .:,         is           .            ssir;;        .rXS: ... ,i....  , ;@@@  
> 
>   ,::         @            :     .      ssrisr;     ,.    :r,... ,. r::,  ,@;@i 
> 
>   ;::        ,@           ..     .      iirrsis;   ..         ...  ,:   .. ,X2r.
> 
>   ;;:.       @,           r      .   .. 2XS25XB5s.    ,,        .. .:        r32
> 
>   rr,,       @           .s     ,.  ... ;,.   .:r:  .: :r           ,          s
> 
>  .;s,:      A;    .      ,.     :  ..       .,:;::,  .. :r   ,    . ..          
> 
>  .ri,:      @     .     .      .:  ,,.: ii@@@@@@@@@s:.   :,  .;   .  ,        ;.
> 
>  .ri::     &.           .      ,,  :.:, 9@@@@@@@@@@@2:,   ;.  ;r   .  .  .    : 
> 
>  ,ri;,.   ;9           ,  .    ;   :.:. @@@@@@@@@@@@@@r   ,,   i:  ..  .  .   .;
> 
>  .rsr.,   @  .        &.  .   :, : ,.:  @@@@@@@@@@@@@@@s  B,   ,,  :,.     .  :9
> 
>  ,rri,:  X;  .       ::      .: r : ,; ,@@@@@@@@@@@@i@@@; @@s:..   :;,    . .   
> 
>  ,sS#:  .2  ,,      .       ,: ;s , :; X@@@@@@@@@@@@ @@@@ 5@@@rr   :r.     .   .
> 
>  :@s   :X   ,      .       .: :@.: .r  @@@@@@@@@@is   .,s::@@@@3,  ;r.     .  :,
> 
>  :. ,@,@,   ,   . .  .    ,r ,#r.. ;; ;@@@@@@;            .r@@@@:  ;r         .:
> 
>   ;@@@@r    ,   r:  .    .s ,3X , :s  @@@@@              ,  , ;@;  rs      :,   
> 
>  .@@@@@,.   ,  rr  .:.   r ,i# . ,s: @@@@: ,     :,      ;G   @@r  ir  .   .i  .
> 
>   @@@@@@,   .  :,. :;.  r..r@   .ii ,@@@@:s  SX.@@3.      5@3  @X  X, .,    r.  
> 
>   @@@@@@G.  ;  ;;:;r;  i, r@.   si  @@@@@@  r@r..   .      @@@@ ;  X ..;    ,;  
> 
>   @@G@@@@5 .:  r;rrr; :; r@,   ri  @@@@@@@r@@@@     @r     @@@@:  .;.,,.    r;  
> 
> ,,2@@  i@@; :  ,:,:r, r i@,   rr  @@@@@@@@@@@@@    ,;@ @s,.@@@@:  ;.S ,   @@@@@ 
> 
> i  @@@:     ,     ,, ; s@    :;..@@@@@@@@@@@@@@@  B@@;@@@  @@@@@r ,sS . r@@; @@@
> 
>  . @@S.      . : .rs,.;#    :;::@@@@@@@@@@@@@@@@,   ,Sr,  @@@r@@  r@. ,@@@:   @@
> 
>  ; :             :2:  X  .A5;:;@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@; i@S  @@@@ @@ ;@
> 
>  .:   . 2rr      ,:  .,.@@;.;@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@5 :@X  .@@@  :@@@@
> 
>   ,  H,M5         :&i;@@@;s@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@;  @s.  @@@  @@@@@@
> 
>  ,. .3i      X     @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@#@@@ i@; ,@; : @@@@ @@@@@@@
> 
>  .  .;@      3:    @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@: @@@@.  ,r  @:@@@@S@@@@@@@@
> 
>     ..@@   @@rs,   @@@@@@@@@@@@@@@ @@@@@@@@@@@@@@@@@@ @@@  .;:,H@2@@@@@@@@@@@@@@
> 
> ,.  : .@@  r@@;;;  @@@@@@@@@@@@@@@ #@@@@@@@@@@@@@@@@@@@@@X@@@@@@@@@@@@@@@@@@@@@@
> 
> ::. 3@ ,@@:   @@@:@@@@@@@@@@@@@@@@@#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ 
> 
>  :;. 9;  G@;#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@.@@@@@@  
> 
>   .;. Ss  ,@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@  3
> 
>     ,.,@@:  .,X@@@@ @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@  :i
> 
>         i@@2;h 3@@@ H@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@5  .s  
> 
> ;,        r,5@ ,@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ ;B: @@, .
> 
> ,:r;;:.   , #A,,##@@@@@@@@@@@@@@@@@@@@B&@@#:       :@@@@@@@@@@@@@@@@@@ @2 @@,   
> 
> .  ,2@@@@@;:Ai@##@@@@@@@@@@@@@@@@@i2@@@@H;.  .,..,  @@@@@@@@@@@@@@@@@@,:r@@    .
> 
> sS:  G@@@@9###M@@@@@@@@@@@@@@@@@   ;:.     ... ...  @@@@@@@@@@@@@@@@@:;@@       
> 
> s :;.:@@@@@@@@@@@@@@@@@@@@@@@@   ..    ,;;riirr;;r @@@@@@@@@@@@@@@@@;.A5  A..  ,
> 
> @ .:. &@@@@@@@@@@@@@@@@@@@@@:  ,,:..:r@@@@@@@@@@@.i@@@@@@@@@@@@@@@@  i5; @@:   X
> 
> @;,::   .@@@@@@@@@@@@@@@@@@@   ,,.;h@@BAHHHHB@H, @@@@@@@@@@@@@@@@@h i5r ,@@    @
> 
> ;@@h:r.. ,@@@@@@@@@@@@@@@@@@@i   S@@@AAAGG32r,,@@@@@@@@@@@@@@@@@@: rrr  @@@ . 5@
> 
>   r@@@@@@@ @@@@@@@@@@@@@@@@@@@@@:.,,:;339G#@@@@@@@@@@@@@@@@@@@@@  ,:;  @@@    @ 
> 
> s    ;ri@@; :@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@. ,:::, .@@.   .  
> 
> i@            @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@.  i@i:.  @@@       
> 
>  ;@;   .       r@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@:   @@ ;,  @@@        
> 
>    @@   r,       r@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@    .@@   @@@@@,       :
> 
>  s, ;@r  sX.        ;@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@2 ;   @@;   @@@@@@@@@@   ;S
> 
>  5s;..;i  ,@9,           ,@@@@@@@@@@@@@@@@@@@@@@@@@,  @@ r@@    @@@@@@@@@@@@@h.,
> 
>                 """)
> 
> 
> 
>                 rem_printed = True
> 
> 
> 
>             elif "rem".lower() not in [waifu.lower() for waifu in waifus]:
> 
>                 rem_printed = False
> 
> 
> 
>             else:
> 
>                 pass
> 
> 
> 
>         case 'exit':
> 
>             exit()
> 
> ```

How is this better than just copy pasting directly into ChatGPT?

**Comment by MementoRC** [comment]
Any updates or timeline for Pycharm accessibility?

**Reply by Rockyread** [reply]
discussions

**Comment by Prateek1009** [comment]
Any updates on the Copilot chat?

**Comment by MementoRC** [comment]
microsoft, nuf said?

**Comment by phucnsp** [comment]
really want copilot chat

**Comment by marcelohweb** [comment]
Any update?

**Comment by bkuepfer-revcomm** [comment]
hello, do we have any updates on this?

**Comment by vlad-ds** [comment]
+1 for Copilot Chat

**Comment by ha-quang-vu** [comment]
+1 please

**Comment by carlosh93** [comment]
Any update for copilot chat in pycharm ? please

**Reply by justinjuner** [reply]
Pleeeeeeeaaaaase cmoooooooooooon +1

**Comment by xmusicxmakerx** [comment]
Yes, please lol. I need this in my life

**Comment by mflilian** [comment]
pleaaaaaseeeeeeeee 😭😭😭😭😭😭😭😭

**Comment by vinilins** [comment]
no update???

**Comment by ferchuni** [comment]
Status?? This is a feature we'd really like to have here.

**Comment by Kev1nDeng** [comment]
Copilot chat for Pycharm please!!!

**Comment by IvanHrytskiv** [comment]
+ 1 Copilot chat for Pycharm please!!!

**Comment by enzoaccelerator** [comment]
Pleaseeeeeeeee any update !?

**Reply by IvanHrytskiv** [reply]
No...((((

**Comment by DiegoArrieta** [comment]
any update with this plugin?

**Comment by bkuepfer-revcomm** [comment]
Hi guys! How is it going? Do we have any updates on this? 
For everyone else, as an alternative, Tab9 does offer a Chat integration with PyCharm.

**Reply by chaoscreater** [reply]
You'd have to be on the Enterprise plan to use that, which seems like a rip off. Rather be using Copilot and pay for the yearly subscription with them instead.

https://www.tabnine.com/pricing

**Reply by bkuepfer-revcomm** [reply]
I agree 100% with you. In short; We need the Copilot chat support on PyCharm asap please! :pray:

**Comment by Zoran-Janjic** [comment]
Hello all, does anyone have any updates on this?

**Comment by nasirovsh** [comment]
Any updates on this. Copilot Chat on Pycharm needed ! Pls

**Comment by MementoRC** [comment]
C'mon, this is microsoft! - Remember Netscape and refrain from getting their beloved "Hall Of Tears" any longer

**Comment by matichek** [comment]
Any updates?

**Comment by GitHub1712** [comment]
1+

**Comment by khamechian1987** [comment]
any update?

**Comment by kj-weber** [comment]
hey there, I just begun my copilot free trial but I will not be continuing my subscription if PyCharm CoPilot chat support is not added.

**Reply by chaoscreater** [reply]
Is that a threat or something? And why would a big corporation like Microsoft care? Not to mention that Github Copilot was never about the Chat feature in the first place. Github Copilot itself works fine in PyCharm. If you're thinking of paying only because of the chat feature, then you have no idea what Github Copilot is for. Go back to using ChatGPT if you want chat so badly.

**Reply by kj-weber** [reply]
I am surprised a random user on a website dedicated to software development would become so upset with a customer giving their honest feedback using a product. I've had the pleasure of meeting the github copilot development team and know they are quite receptive to suggestions for adding support or features, so I hope my datapoint helps their product managers determine priorities. 

You made a lot of false assumptions in your hostile response here, I have simply tried copilot with chat, and tried it without, and I personally believe the former is the only product worth the price they are asking me to pay as a consumer. I'm not really interested in bickering with random community members.

**Reply by Megasteel32** [reply]
@chaoscreater microsoft isnt going to give you monies for sucking them off on github discussions

**Reply by chaoscreater** [reply]
Looks like someone got triggered.

It's really simple. Don't pay for the product if you don't like it. Don't act like a Karen by making statements like "if something isn't to my satisfaction, I won't pay for it". Demanding for something is childish. Microsoft doesn't care if they miss out on your $150 annual subscription for Copilot.

The funny thing is, Github Copilot was NEVER about the chat feature in the first place. If the chat feature was never announced, you would never have something to complain about. 

Go cry some more.

**Reply by Megasteel32** [reply]
who the fuck are you again?

**Reply by chaoscreater** [reply]
Who the fuck am I? Someone who obviously triggered you :)

https://imgflip.com/i/82m0fh

**Reply by kj-weber** [reply]
I have cancelled my copilot free trial, I will not be paying for this for the reason listed above.

**Reply by MementoRC** [reply]
FYI, `Sourcery`  https://github.com/sourcery-ai/sourcery is a very satisfactory alternative, if not a significant upgrade.

**Comment by sebastien-moha** [comment]
+1 

Pycharm Copilot Chat will be very useful !

**Reply by dereklarmstrong** [reply]
+1 copilot chat in pycharm!!!

**Comment by yassinelaouni** [comment]
respense#1

**Comment by itsDrew86** [comment]
+1 for pycharm support. I will probably discontinue my copilot sub until chat is available on pycharm.

**Comment by samperay** [comment]
+1 for pycharm copilot chat support, please provide us at the earliest

**Comment by Stephane-Perriot** [comment]
+1, come on guys ! please :)

**Comment by shahabejaz** [comment]
+1 for copilot chat!

**Comment by dariashekhovtsova-hs** [comment]
+1 for copilot chat in PyCharm

**Comment by yanxunhu** [comment]
+1 for copilot chat in PyCharm

**Comment by rgs258** [comment]
+1. Good stuff here; thanks copilot team!

**Comment by caliph007** [comment]
+1 for copilot chat in PyCharm

**Comment by MementoRC** [comment]
`Sourcery` for the win

**Reply by aitirga** [reply]
+1 for copilot chat!

**Comment by auideaslab** [comment]
copilot chat please

**Comment by jpang32** [comment]
+1 for copilot chat in PyCharm

**Comment by zycliao** [comment]
Please bring copilot chat to Pycharm! This would be really helpful!

**Comment by pretty1020** [comment]
Co Pilot chat in Pycharm please!

**Comment by 9p15p** [comment]
Copilot chat in Pycharm please!

**Comment by mastrolube** [comment]
please :(

**Comment by ghost** [comment]
yes, add this feature ASAP

**Comment by BRNNDR** [comment]
+1 for Copilot Chat support in Pycharm

**Reply by ihormaslovnix** [reply]
+1 for Copilot Chat support in Pycharm

**Comment by Akash1134** [comment]
👋  We are happy to announce that a private beta of GitHub Copilot Chat is now available for users of JetBrains IDEs, including IntelliJ, PyCharm, WebStorm, Android Studio, and more. 🚀 

**This private beta is available to Copilot Business customers and Copilot Individual users.**

### To get access to the private beta, [sign up for this waitlist](https://github.com/github-copilot/chat_jetbrains_waitlist_signup/join).

GitHub Enterprise Cloud (GHEC) administrators interested in participating in the private beta should reach out to your GitHub account manager or [contact our sales team](https://github.com/enterprise/contact) to make the feature available for your enterprise.

We’d love your feedback on this new release. Please [use this link to share your feedback](https://gh.io/copilot-chat-jb-feedback) or ideas on how to improve the product. Thank you for your patience and support.

**Reply by filias** [reply]
I have tried to sign up and I am a Business user but I see this:
![Screenshot 2023-11-10 at 11 01 11](https://github.com/community/community/assets/853564/b167c3f3-56b3-4e25-b30a-4b234211352b)
We do have co-pilot enabled for the organization and we did enable `GitHub Copilot Chat for IDEs` in the settings.
Any idea what can be the problem?

**Reply by trosemidmark** [reply]
I am in the same situation. Seems fine in VSCode, but does not work in PyCharm or IntelliJ IDEA

**Reply by peter-kovacs-aimotive** [reply]
Same here with PyCharm.

**Reply by ghost** [reply]
Same here

**Reply by ramazan-safiullin** [reply]
They made it on purpose to force users switch to vscode, I'm not surprised, it's Microsoft, they always use dirty competition methods

**Reply by marton-helli-aimotive** [reply]
to me it finally started working with today's update, 1.4.19.5096

**Comment by andreagrandi** [comment]
@Akash1134 I work with @filias and this is what I see in our organisation settings for Copilot:
![Screenshot 2023-11-10 at 13 30 45](https://github.com/community/community/assets/636391/cab99c5f-a5c5-40b3-9f70-6fc353e02e8a)
What else are we missing?

p.s: not sure if it can be useful, but my account is the only one with access to Copilot Chat, but it must be noted that I separately requested to join the preview release list a few months ago, so I may have the access for that reason. Other colleagues (part of the same organisation) don't have it instead.

**Reply by mkx-pm** [reply]
Same here. We have copilot chat enabled for the entire org. But I still don't see any options for Jetbrains IDEs.

**Reply by prismec** [reply]
I have a similar issue. GitHub Copilot Chat is enabled for our organisation and thus also for my user. It also works for Visual Studio Code, but IDEA still shows this screen

<img width="601" alt="image" src="https://github.com/community/community/assets/3080401/14e2a4c1-6a7d-4851-a19c-8826bfa3a839">

**Reply by itsDrew86** [reply]
+1 for open beta

**Reply by francisconroy** [reply]
Same issue for us in the SwitchDin organisation, how can we get access to the private beta waitlist?

**Reply by odesiuz** [reply]
I don't think copilot chat is fully functional for pycharm. It is fully functional in IntelliJ IDEA. Switching might be a temporary solution pending when it's fully functional in pycharm

**Reply by odesiuz** [reply]
Update Pycharm or Jetbrains IDE to the newer version and opt to use the new UI version. With the new UI version, you can see the copilot chat on the right side

**Reply by andreagrandi** [reply]
@odesiuz as you can see from https://github.com/orgs/community/discussions/56901#discussioncomment-7587118 message, the problem is not that Copilot Chat UI doesn't show up in Pycharm. The problem is that it asks you to join the wait list even if you have enabled it at organisation level.

**Reply by Stihotvor** [reply]
Do you happen to have any updates on the chat+PyCharm issue? I've started to use Copilot at work and it works well, but without Chat it seems, in most cases, a free AWS alternative does more. I would like to switch to Your solution the same day, you fix the integration issue.

**Reply by sanjayhallan** [reply]
same issue for me, enabled via organisation but still get message to join waiting list on pycharm

**Reply by GabrielBoninUnity** [reply]
same here with the org

**Reply by Riofd** [reply]
I still can't use Copilot Chat in Pycharm until 1/2024.

**Reply by plysytsya** [reply]
When is it going to be released for all users?

**Reply by AhmedTooHere** [reply]
08.02.24 and still not there.

**Reply by jdoe-id1** [reply]
19.02.24 and still no chat :(

**Reply by luggie** [reply]
jesus ...

**Comment by lekepi** [comment]
I have received the email saying that: "You've been granted access to GitHub Copilot Chat in JetBrains IDEs private beta." But when I go to Setting->copilot-> policies as suggested, there is no option to add the chat.

**Comment by aj0424754** [comment]
Hi, we are happy to be in the private Beta, and are using GitHub Copilot Chat for Business in an EMU organization with JetBrains IDE and it works like a Charm. 
Thank you GitHub team !

**Comment by Akash1134** [comment, marked as answer]
Following our [Private Beta](https://github.blog/changelog/2023-11-08-copilot-in-jetbrains-ides-chat-is-now-available-in-private-beta-new-welcome-guide/), we are thrilled to announce **Copilot Chat in JetBrains IDEs is now generally available (GA)** 🎉

Driven by GPT-4, GitHub Copilot Chat provides instant guidance directly within various JetBrains IDEs, such as PyCharm, IntelliJ IDEA, WebStorm, Rider, and more. This contextually-aware tool tailors suggestions to your specific coding tasks and even allows explicitly adding files for reference. It empowers developers to innovate efficiently by assisting with complex concepts, code explanations, unit testing, and many more use cases, all while effortlessly adjusting to your preferred language style. 

You can learn more about the release in this [changelog](https://github.blog/changelog/2024-03-07-github-copilot-chat-general-availability-in-jetbrains-ide/). 

If you've got any questions or feedback to share, please post them in [this discussion](https://github.com/orgs/community/discussions/110614).

**Reply by mathijsfr** [reply]
How do you get copilot chat in your IDE. Just like the OP I still only see Github Copilot completions in my Pycharm sidebar.

**Reply by ZTaylor12** [reply]
> How do you get copilot chat in your IDE. Just like the OP I still only see Github Copilot completions in my Pycharm sidebar.

I have a tab on the far right called Github Copilot Chat, next to other tabs such as Notifications and Python Console.
